platform-sdk/docs/proposals/001-tss-rosters-design-proposal/001-tss-rosters-design-proposal.md

```diff
@@ -0,0 +1,265 @@
+management for TSS operations.
+This proposal attempts to provide a specification for the behavior of rosters starting from their creation from the CAB
+to their terminal state within the TSS specification.
+
+### Requirements
+
+- Immutability: Rosters must be immutable to ensure the integrity of the computed hash.
+- Efficient Storage: Roster data should be stored efficiently in the state to minimize overhead and avoid unnecessary
+  data copying.
+- Clear API: A well-defined API should be provided for the Hedera app to create and submit candidate rosters, and for
+  the platform to manage the lifecycle of submitted rosters.
+- Reconnect Compatibility: The reconnect process must be adapted to use rosters instead of address books for validation
+  and node selection.
+- Address book paradigm discontinued within the platform codebase, only use rosters instead
+
+### Changes
+
+### Architecture and/or Components
+
+###### Roster API
+
+The Hedera app is already responsible for managing the address book. We propose that it continues to do so, and at
+adoption time, creates a candidate Roster object from the CAB and set it in the state. The State will contain one map of
+rosters, keyed by roster’s hash - ```Map<Hash<Roster>, Roster>```. It should be noteworthy that the roster must be
+immutable so as to guarantee the integrity of the computed hash. This map of rosters will typically contain the
+currently active roster, the optional previously active roster, and an optional candidate roster. There will be new
+fields in PlatformState - `candidateRosterHash` and `activeRosterHash` - such that at adoption time, the way to trigger
+the adoption of a roster will be by the client code inserting a roster in the roster map, alongside setting the
+candidateRosterHash field in the PlatformState. The `activeRosterHash` will be private to the platform and will not be
+settable or modifiable from outside the platform.
+
+This indirection avoids moving data around in the merkle state which requires data to be copied into the block stream.
+This is computationally expensive. Another benefit of this approach is that adoption trigger becomes very simple (app
+sets the roster) with clear responsibility between the app and the platform.
+
+Some edge cases worth considering include:
+
+1. multiple concurrent roster submission - has a potential for introducing race conditions. We will prevent this by
+   guaranteeing immutability on rosters.
+2. Size control on the map of rosters - we certainly don’t want this map to grow infinitely so insertion of rosters will
+   be controlled, with clear rules for removal of unused rosters (e.g. the acceptance of a new candidate roster
+   invalidates and removes existing one).
+
+###### Roster API - Implementation
+
+Implementation of some new TSS components has been proposed.
+See [link](https://www.notion.so/TSS-Platform-Architecture-04b15df371ba4b1d848360542e05a030?pvs=21).
```

Comment:
Technically, the TSS-Roster is unlikely to need any changes in the components or wiring. At least not the ones currently described in this Notion page. So I suggest to remove these two lines because the TSS-Ledger-Id proposal is responsible for these changes.